Port px4vision airframe to gz environment

TheotimeBalaguer opened the pull request:
<!--

Thank you for your contribution!

Get early feedback through
- Dronecode Discord: https://discord.gg/dronecode
- PX4 Discuss: http://discuss.px4.io/
- opening a draft pr and sharing the link

-->

### Solved Problem
"PX4 Vision" Airframe availability for simulation in Gazebo

Fixes #{Github issue ID}

### Solution
- Port the .sdf file from the model existing in gazebo-classic

### Test coverage
- Gazebo-Garden test : With the launch from the command-line and control with QGroundControl : no problem
- Gazebo-Garden multidrone : I launched a simulation with 4 PX4Vision models and everything seems to work
- No Unit/integration tests.

### Context
Related links, screenshot before/after, video

beniaminopozzan commented:
Hi @TheotimeBalaguer , how did you start your simulations/tests? The usual
```sh
make px4_sitl gz_px4vision
```
doesn't work. Which is expected as you did not define an suitable airframe for that model.

Even spawning the model manually from the Gazebo GUI doesn't work:

```sh
Error [SDF.cc:154] Tried to use callback in sdf::findFile(), but the callback is empty.  Did you call sdf::setFindCallback()?
Error [parser.cc:748] Error finding file [/home/pozzan/git/PX4-Autopilot-gz/Tools/simulation/gz/models/px4vision/px4vision.sdf].
```

TheotimeBalaguer commented:
Thanks for the review Beniamino,
Actually, I directly launch the `px4` binary with `export PX4_GZ_MODEL=px4vision` while Gazebo is running. :smile: 
I have to admit that I am a bit lost on the compilation workflow of the PX4 project, is there any existing docs about that ?

What I found : the targets for Gazebo simulation are defined in `Tools/simulation/gz/gz_bridge/CMakeLists.txt` based upon the airframes defined in `ROMFS/px4fmu_common`.
Thus, I created a file "4006_gz_px4vision" which is nothing more than a mix between `ROMFS/px4fmu_common/init.d/airframes/4020_holybro_px4vision_v1_5` and `ROMFS/px4fmu_common/init.d-posix/airframes/4001_gz_x500`. 
I gave the ID 4006 kind of randomly because it is the 6th model to be added to gz simulation, but also hesitated with 4020 as it is its ID in init.d...
@LorenzMeier could you please confirm I did nothing wrong with the naming ? (I think you are the one managing airframes).

Now, the command `make px4_sitl gz_px4vision` works, but I think a verification of the airframe is needed (file `4006_gz_px4vision`), both for name and content. In particular, I commented some lines that were not recognized at the start of the PX4.

Thanks all for the help ! :+1:

Jaeyoung-Lim commented:
@TheotimeBalaguer Since this contains mesh files and is redundant to the x500_depth model, Would it be possible to put this up on ignition fuel?

Jaeyoung-Lim commented on Tools/simulation/gz/models/px4vision/model.config:
```diff
@@ -0,0 +1,15 @@
+
+  <author>
+   <name>Theotime Balaguer</name>
+   <email>theotime.balaguer@insa-lyon.fr</email>
```
Isn't this model a port from an existing model?

TheotimeBalaguer commented:
> @TheotimeBalaguer Since this contains mesh files and is redundant to the x500_depth model, Would it be possible to put this up on ignition fuel?

Sure, but the best would be to upload it with the PX4 account on fuel, like the x500 model. I don't know who manages this kind of things.

TheotimeBalaguer commented:
@Jaeyoung-Lim my Ignition Fuel username is Tbalaguer. Thanks !

Jaeyoung-Lim commented:
@TheotimeBalaguer Done! Let me know if you can upload the model!
If this is done, we can link the model directly from this repo to the model

Thanks!

TheotimeBalaguer commented:
@Jaeyoung-Lim Thank you, I imported the px4vision model to the PX4 organization and changed to a Fuel import in the .sdf file. I removed the meshes to reduce a bit the size of the PX4 project.

Jaeyoung-Lim reviewed (approved):
LGTM, Thanks!
